pinot-common/src/main/java/org/apache/pinot/common/function/scalar/StringFunctions.java

```diff
@@ -570,6 +572,81 @@ public static String[] split(String input, String delimiter, int limit) {
     return StringUtils.splitByWholeSeparator(input, delimiter, limit);
   }
 
+  /**
+   * @param input an input string for prefix strings generations.
+   * @param length the max length of the prefix strings for the string.
+   * @param regexChar the character for regex matching to be added to prefix strings generated. e.g. '^'
+   * @return generate an array of prefix strings of the string that are shorter than the specified length.
+   */
+  @ScalarFunction
+  public static String[] prefix(String input, int length, String regexChar) {
+    ObjectSet<String> prefixSet = new ObjectLinkedOpenHashSet<>();
```

Comment:
Missed this in the first pass. Can we return a list instead since that seems more fitting for `prefixes`?

For the set version we could create `uniquePrefixes, uniqueSuffixes` or something. Moreover we should specify how the set will be ordered since otherwise the replicas of consuming segments will diverge.

Reply:
this is a very great point, but i think LinkedHashSet will maintain the same order when item have been added to the set.

Reply:
I see. But it will only return the unique prefixes. Can we rename the method to `uniquePrefixes` or similar to be explicit about this?

`prefixes` suggests that we will enumerate all prefixes.

Reply:
`uniquePrefixes` sounds good. let me rename it.